### DISCUSSION TITLE
[FEATURE REQUEST] Update Venafi Runner to support token-based authentication

### DISCUSSION THREAD

**Discussion by paulternate** [saltstack/salt#67483] [Feature Requests]
### Description of Issue
Token-based authentication provides a better, token-based approach to authenticate to Venafi Trust Protection Platform. This limits the wide scope of the old API-key authentication method, but does require some additional initial configuration. 

In addition to the security benefits token-based authentication provides, it also mitigates the need to re-authenticate before every call, since the auth token can be longer lived. I’d be happy to answer questions or walk through the setup if it would be beneficial.
 
The API-Key authentication method has already been removed from the official documentation (although it is still functional). Integrations still using API-Key based authentication will not be compatible with Venafi Trust Protection Platform after this functionality is deprecated. 

### Setup
I am happy to provide access to documentation, a Venafi testing instance, as well as be available for any questions that might arise during the change.

### Versions Report
Affects all versions.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://saltstackcommunity.herokuapp.com/)
  - [IRC on Freenode](https://webchat.freenode.net/#salt)
  - [SaltStack YouTube channel](https://www.youtube.com/user/SaltStack)
  - [SaltStackInc Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at core@saltstack.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!
